pymeasure/instruments/keithley/keithley2281S.py

```diff
@@ -0,0 +1,345 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2024 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import enum
+
+import numpy as np
+import pandas as pd
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.keithley.buffer import KeithleyBuffer
+from pymeasure.instruments.validators import strict_discrete_set, strict_range, truncated_range
+
+
+@enum.unique
+class Keithley2281SOperationCondition(enum.IntFlag):
+    """Enum containing Keithley2281S operation condition definition"""
+    CALIBRATION = 1
+    _RESERVED_1 = 2
+    _RESERVED_2 = 4
+    _RESERVED_3 = 8
+    MEASUREMENT = 16
+    TRIGGER = 32
+    ARM = 64
+    _RESERVED_4 = 128
+    FILT = 256
+    LIST = 512
+    IDLE = 1024
+    _RESERVED_5 = 2048
+    _RESERVED_6 = 4096
+    _RESERVED_7 = 8192
+    _RESERVED_8 = 16384
+
+
+class Keithley2281S(Instrument, KeithleyBuffer):
+    """
+    Represents the Keithley 2281S-20-6 power supply and battery simulator / characterizer.
+    Common commands beside `function_mode` and power supply commands should also work for
+    Keithley 2280S power supplies, although this is untested.
+    """
+    _VOLTAGE_RANGE = [0.0, 20]
+    _CURRENT_RANGE_PS = [0.1, 6.1]
+    _CURRENT_RANGE_BT_BS = [0.0, 6.1]
+    _INTERNAL_MEMORY_SLOTS = [*range(1, 10)]
+
+    def __init__(self, adapter, name="Keithley2281S", **kwargs):
+        super().__init__(adapter, name, includeSCPI=True, **kwargs)
```

Comment:
I'm sorry for the back and forth:
Recently I told you to specify `includeSCPI` explicitly, which you did, thank you.
That was, however, only in preparation for a larger change (#905).
Now, that other PR has been merged, so the new style to define SCPI compliant instruments works differently:

```suggestion
from pymeasure.instruments import SCPIMixin

class Keithley2281S(SCPIMixin, Instrument, KeithleyBuffer):
    """
    Represents the Keithley 2281S-20-6 power supply and battery simulator / characterizer.
    Common commands beside `function_mode` and power supply commands should also work for
    Keithley 2280S power supplies, although this is untested.
    """
    _VOLTAGE_RANGE = [0.0, 20]
    _CURRENT_RANGE_PS = [0.1, 6.1]
    _CURRENT_RANGE_BT_BS = [0.0, 6.1]
    _INTERNAL_MEMORY_SLOTS = [*range(1, 10)]

    def __init__(self, adapter, name="Keithley2281S", **kwargs):
        super().__init__(adapter, name, **kwargs)
```
Please move the import together with the import of `Instrument`.

This mixin-based implementation is a first step to create more harmonized instruments.